tests/test_permutation.py

```diff
@@ -9,12 +9,12 @@ def test_init(self):
         with self.assertRaises(AssertionError): Permutation([2,1,2], check=True)
         with self.assertRaises(AssertionError): Permutation([1,1], check=True)
         with self.assertRaises(AssertionError): Permutation([2], check=True)
-        with self.assertRaises(AssertionError): Permutation(set([1,2,3]), check=True)
         with self.assertRaises(AssertionError): Permutation(5, check=True)
         with self.assertRaises(AssertionError): Permutation(None, check=True)
         Permutation([], check=True)
         Permutation([1], check=True)
         Permutation([4,1,3,2], check=True)
+        Permutation(set([1,2,3]), check=True)
```

Comment:
Agreed, it is unfortunate; using a set to initialize a permutation is dumb. However, you should be able to do it. It is consistent with how tuples behave:
```
>>>tuple(set([1,2,3,4]))
(1, 2, 3, 4)
```
and we'll be making permutations a subclass of tuples in the near future.
At most we could have some sort of warning, but I think we are safe to assume some python aptitude from users of the permutation class.